common/timing.cc

```diff
@@ -811,6 +812,80 @@ void timing_analysis(Context *ctx, bool print_histogram, bool print_fmax, bool p
     }
 
     if (print_path) {
+        static auto print_net_source = [](Context *ctx, NetInfo *net) {
+            auto sources = net->attrs.find(ctx->id("src"));
+            if (sources == net->attrs.end()) {
+                // No sources for this net, can't print anything
+                return;
+            }
+
+            // Sources are separated by pipe characters, deepest source last
```

Comment:
It's not a *sequence* of sources but an *unordered set* of sources; that's how Yosys treats the `src` attribute internally:
![Screenshot_20200830_221155](https://user-images.githubusercontent.com/54771/91670509-d0e51e00-eb0d-11ea-82cc-1b03141a517e.png)